File: src/backend/src/models/user_details.go
package models

// a deeper view of a user
type GetUserDetailsReply struct {
	ID             Snowflake      `json:"id"`
	Username       string         `json:"username"`
	Discriminator  string         `json:"discriminator"`
	Avatar         string         `json:"avatar"`
	Goals          []PersonalGoal `json:"goals"`
	CurrentStreak  DayStreak      `json:"current_streak"`
	LongestStreak  DayStreak      `json:"longest_streak"`
	LastActivities []Sport        `json:"last_activities"`
}

File: src/frontend/src/utils/api/responses/UserDetails.ts
import { StreakData } from '../../../models/Streak';
import { Sport } from '../../../models/Sport';
import { PersonalGoalData } from './PersonalGoals';

export interface UserDetails {
  current_streak: StreakData;
  longest_streak: StreakData;
  goals: PersonalGoalData[];
  last_activities: Sport[];
}

export interface GetUserDetailsResponse {
  id: number;
  username: string;
  discriminator: string;
  avatar: string;
  current_streak: StreakData;
  longest_streak: StreakData;
  goals: PersonalGoalData[];
  last_activities: Sport[];
}
